### DISCUSSION TITLE
Wrong error reporting on multiple GeneratedValue annotation usage

### DISCUSSION THREAD

**Discussion by Genarito** [doctrine/doctrine2#8993] [Support Questions]
Hello, I don't know if the issue is still present in the current versions of Doctrine. The version I'm using is `doctrine/orm": "2.3.x-dev` and, as I describe in [this Stack Overflow question](https://stackoverflow.com/q/69036275/7058363) when `GeneratedValue` is used on multiple fields with different strategies Doctrines uses only the last definition. For instance, defining: 

```php
/**
 * @var integer
 *
 * @ORM\Column(name="CONTRACT", type="integer", nullable=false)
 * @ORM\Id
 * @ORM\GeneratedValue(strategy="SEQUENCE")
 * @ORM\SequenceGenerator(sequenceName="SID_CONTRACT", allocationSize=1, initialValue=1)
 */
private $contract;

/**
 * @var string
 *
 * @ORM\Column(name="TYPE", type="string", length=1, nullable=false)
 * @ORM\GeneratedValue(strategy="NONE")  // This was the conflicting line
 */
private $type;
```

Will wire the `Doctrine\ORM\Id\AssignedGenerator` to the field `contract` instead of  `Doctrine\ORM\Id\SequenceGenerator`. **More detailed information in my S.O. question**.

Maybe it'll be great to improve the documentation to describe this behavior or fix it if it's not expected.

Thanks!

**Comment by greg0ire** [comment]
> The version I'm using is `doctrine/orm": "2.3.x-dev"`

Why are you not using a stable version? That area of the code has been touched lately, so it's quite important to determine which version you are actually using with `composer show|grep orm`. Also, did you suddenly experience this issue after an upgrade?

**Comment by derrabus** [comment]
Let's move this to support. I don't think that an issue with a more than seven years old dev snapshot counts as a bug report. 🙂

**Reply by Genarito** [reply]
Hi @greg0ire and @derrabus,

I've updated `doctrine/orm` to `2.4.8` version and the bug is still present. I can't check with the problem persists in newer versions as I'm working on a legacy project

**Reply by greg0ire** [reply]
@derrabus if you try using the version constraint above right now, it will install 2.10.x, that's why I asked. But now that we are sure, indeed, it should be a discussion.
